text/0000-maybe-dangling.md

```diff
@@ -0,0 +1,247 @@
+# `maybe_dangling`
+
+- Feature Name: `maybe_dangling`
+- Start Date: 2022-09-30
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Declare that references and `Box` inside a new `MaybeDangling` type do not need to satisfy any memory-dependent validity properties (such as `dereferenceable` and `noalias`).
+
+# Motivation
+[motivation]: #motivation
+
+### Example 1
+
+Sometimes one has to work with references or boxes that either are already deallocated, or might get deallocated too early.
+This comes up particularly often with `ManuallyDrop`.
+For example, the following code is UB at the time of writing this RFC:
+
+```rust=
```

Comment:
```suggestion
```rust
```

And the same in the rest of the document.

Reply:
Ah that seems to be a non-standard markdown extensions of hackmd...